File: app/src/main/java/com/chooloo/www/callmanager/ui/main/MainMvpPresenter.java
package com.chooloo.www.callmanager.ui.main;

import android.content.Intent;
import android.view.MenuItem;

import com.chooloo.www.callmanager.ui.base.MvpPresenter;

public interface MainMvpPresenter<V extends MainMvpView> extends MvpPresenter<V> {

    void onDialpadFabClick();

    void onSearchTextChanged(String text);

    void onSearchFocusChanged(boolean isFocused);

    void onDialNumberChanged(String number);

    void onOptionsItemSelected(MenuItem item);

    boolean onBackPressed();

    void onMenuClick();

    void handleViewIntent(Intent intent);
}

File: app/src/main/java/com/chooloo/www/callmanager/ui/main/MainMvpPresenter.kt
package com.chooloo.www.callmanager.ui.main

import android.content.Intent
import android.view.MenuItem
import com.chooloo.www.callmanager.ui.base.MvpPresenter

interface MainMvpPresenter<V : MainMvpView> : MvpPresenter<V> {
    fun onDialpadFabClick()
    fun onSearchTextChanged(text: String)
    fun onSearchFocusChanged(isFocused: Boolean)
    fun onDialNumberChanged(number: String)
    fun onOptionsItemSelected(item: MenuItem)
    fun onBackPressed(): Boolean
    fun onMenuClick()
    fun onViewIntent(intent: Intent)
}